### PR TITLE
Issue 1344: Forbid `float` and `complex` compares

### LINKED ISSUE #1344: Forbid compare with float
# Rule request

## Thesis

This code might not work as one would think:

```python
x == 3.0
```

Because `x` can be `0.3 / 0.1` which is:

```
>>> 0.3 / 0.1
2.9999999999999996
```

## Reasoning

This is a best practice rule.

### PR DESCRIPTION
**Pull request by raghul-rajasekar**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #1344 

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: tests/fixtures/noqa/noqa.py [hunk 1]
```diff
@@ -711,3 +711,8 @@ def consecutive_yields():
         # https://github.com/wemake-services/wemake-python-styleguide/issues/1082
         break
     my_print(4)
+
+f1 = 3.0
```
Comment (sobolevn) [inline tests/fixtures/noqa/noqa.py:715]: You can use just `my_print(some_var == 3.0)` 
We tend to use the simplest example in this file.

File: tests/test_visitors/test_ast/test_compares/test_float_complex_compare.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+@pytest.mark.parametrize('code', [
+    'x > 3',
+    'x <= y',
+    'abs(x - y) <= eps',
+    'isclose(x, 5.0)',
+    'isclose(y, 3j)',
+    '3 in item_list',
+    '3 != 5',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_compares/test_float_complex_compare.py:45]: Let's also add `3.0 + x` case

File: wemake_python_styleguide/violations/best_practices.py [hunk 3]
```diff
@@ -2106,3 +2108,38 @@ class ShebangViolation(SimpleViolation):
...
+        Use fuzzy operators.
+        1. ``abs(f1 - f2) <= allowed_error``
+        2. ``math.isclose(f1, f2)`` (for ``float``)
+        3. ``cmath.isclose(c1, c2)`` (for ``complex``)
+        4. Custom logic, not using operators
+
+    Example::
+
+        # Correct:
+        math.isclose(3.0, 0.3/0.1)
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py:2133]: Please, format these examples to follow `pep8`: `math.isclose(3.0, 0.3 / 0.1)`

Comment (sobolevn) [reply wemake_python_styleguide/violations/best_practices.py:2133]: And so on

File: wemake_python_styleguide/violations/best_practices.py [hunk 4]
```diff
@@ -2106,3 +2108,38 @@ class ShebangViolation(SimpleViolation):
...
+
+        # Wrong:
+        3.0 == 0.3/0.1
+        3+4j == (0.3+0.4j)/0.1
+
+    .. versionadded:: 0.14.0
+
+    """
+
+    error_template = 'Found comparison with float/complex'
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py:2144]: `Found comparison with float or complex number`

**Comment by raghul-rajasekar** [comment]
Great! This is my first open-source contribution. Thanks a lot for your guidance!

**Comment by sobolevn** [comment]
@raghul-rajasekar congrats on that 🎉 
This is really an awesome feeling, I remember it quite well! 

Feel free to participate in any other issue!

**Pull request merged**